docs/theming.md

```diff
@@ -109,15 +109,91 @@ dark theme.
 In order to style your own components with our tooling, the component's styles must be defined 
 with Sass. 
 
-You can consume the theming functions and variables from the `@angular/material/core/theming`.
-You can use the `md-color` function to extract a specific color from a palette. For example:
+You can consume the theming functions and variables from the @angular/material/core/theming. You can use the `map-get` function to extract the theming variables and `md-color` function to extract a specific color from a palette. For example:
+
+app/custom-input/custom-input-theme.scss
+
 ```scss
-.unicorn-carousel {
-  background-color: md-color($primary);
-  color: md-color($primary, default-contrast);
+@import '~@angular/material/core/theming/theming';
+
+@mixin custom-input-theme($theme) {
```

Comment:
@jelbourn I've placed this much of css because I was trying to help people creating their own form elements that use all input styles, from `placeholder-color` to `underline-focused-color`. Maybe I can point the `_input-theme.scss` as a base to customize inputs and show only this tiny format that you suggested. What do you think?
Or you are saying to me that we can use this tiny format in `theming.md` and in the new file called `theming-your-components.md` we place that much of code that I did in this PR?